File: app/src/main/java/android1601/itstep/org/kidsgame/program/enums/POSITION_TYPE.java
package android1601.itstep.org.kidsgame.program.enums;

/**
 * Created by roman on 03.04.2017.
 */

public enum POSITION_TYPE {
    WATER(3),AIR(2),GROUND(1),UNKNOWN(0);
    private int position;

    POSITION_TYPE(int position) {
        this.position = position;
    }

    public static POSITION_TYPE fromPositions(int value){
        for (POSITION_TYPE type: POSITION_TYPE.values()) {
            if (type.getPosition()==value){
                return type;
            }
        }
        return UNKNOWN;
    }

    public int getPosition() {
        return position;
    }
}

File: app/src/main/java/android1601/itstep/org/kidsgame/program/enums/POSITION_TYPE.kt
package android1601.itstep.org.kidsgame.program.enums

/**
 * Created by roman on 03.04.2017.
 */

enum class POSITION_TYPE private constructor(val position: Int) {
    WATER(3), AIR(2), GROUND(1), UNKNOWN(0);


    companion object {

        fun fromPositions(value: Int): POSITION_TYPE {
            for (type in POSITION_TYPE.values()) {
                if (type.position == value) {
                    return type
                }
            }
            return UNKNOWN
        }
    }
}
